### PR TITLE
Restructure the metadata output format

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Produces an object, keyed on test path, instead of an array.
Includes subtest and status filtering for each link.

### CODE REVIEW

File: webapp/components/test-file-results.js [hunk 1]
```diff
@@ -273,7 +277,8 @@ class TestFileResults extends WPTFlags(LoadingState(PathInfo(
     return new Map([...firstScreenshot, ...Object.entries(screenshots)]);
   }
 
-  computeRows(resultsTable, onlyShowDifferences) {
+  computeRows(resultsTable, metadata, onlyShowDifferences) {
+    console.log(resultsTable, metadata);
```
Comment (lukebjerring) [inline webapp/components/test-file-results.js]:
```suggestion
```

File: webapp/components/test-file-results.js [hunk 2]
```diff
@@ -73,13 +73,17 @@ class TestFileResults extends WPTFlags(LoadingState(PathInfo(
         type: Array,
         value: [],
       },
+      metadata: {
+        type: Array,
```
Comment (lukebjerring) [inline webapp/components/test-file-results.js]:
```suggestion
        type: Object,
```

File: webapp/components/test-file-results.js [hunk 3]
```diff
@@ -73,13 +73,17 @@ class TestFileResults extends WPTFlags(LoadingState(PathInfo(
...
+        value: [],
```
Comment (lukebjerring) [inline webapp/components/test-file-results.js]:
```suggestion
        value: {},
```

File: webapp/views/wpt-results.js [hunk 4]
```diff
@@ -357,6 +358,7 @@ class WPTResults extends WPTColors(WPTFlags(PathInfo(LoadingState(TestRunsUIBase
         type: Array,
         computed: 'computeDisplayedTests(path, searchResults)',
       },
+      metadata: Array,
```
Comment (lukebjerring) [inline webapp/views/wpt-results.js]:
```suggestion
      metadata: Object,
```

**Comment by lukebjerring** [comment]
@jgraham - might be worth getting your input on the data format here, since you'll be a consumer.

**Comment by jgraham** [comment]
The final format is something like

```
{
  <test_id>: 
  [
    {
      "product": <product>,
      "url": <url>,
      "results": 
      [
        {
          "status": <status>, 
          "subtest": <subtest>
        }
      ]
    }
  ]
} 
```
?

That seems OK, I think. Arguably it could be keyed on `test, subtest` with subtest being `null` for links with top-level tests, but I guess that's akward in languages with only string keys. Bundling together `status` and `subtest` seems a little odd, but none of this is a blocker; a lookup by test name followed by a scan of annotations for that test seems like it's going to be necessary regardless of the details of the format.

File: api/azure/gomock_reflect_313602269/prog.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
+
```
Comment (KyleJu) [inline api/azure/gomock_reflect_313602269/prog.go]: What is this file?

File: webapp/views/wpt-app.js [hunk 6]
```diff
@@ -133,7 +133,8 @@ class WPTApp extends PathInfo(WPTFlags(TestRunsUIBase)) {
                      path="{{subroute.path}}"
                      test-runs="{{testRuns}}"
                      test-paths="{{testPaths}}"
-                     search-results="{{searchResults}}"></wpt-results>
+                     search-results="{{searchResults}}"
+                     metadata="[[metadata]]"></wpt-results>
```
Comment (KyleJu) [inline webapp/views/wpt-app.js]: metadata is fetched inside of wpt-metadata.js so it should be always empty at this level?

**Review by nickeldennis1980 — APPROVED** [review]

**Pull request merged**